packages/cli/configuration/src/generators-yml/addGenerator.ts

```diff
@@ -7,8 +8,70 @@ import {
 } from ".";
 import { GENERATOR_INVOCATIONS } from "./generatorInvocations";
 
-export function upgradeGenerator() {
-    // TODO
+function getOrThrowGeneratorName(generatorName: string, context: TaskContext): GeneratorName {
+    const normalizedGeneratorName = normalizeGeneratorName(generatorName);
+    if (normalizedGeneratorName == null) {
+        return context.failAndThrow("Unrecognized generator: " + generatorName);
+    }
+
+    return normalizedGeneratorName;
+}
+
+async function getLatestGeneratorVersion(generatorName: string): Promise<string> {
+    const docker = new Docker();
+    const image = await docker.listImages({
+        filters: JSON.stringify({
+            reference: `${generatorName}:latest`
+        })
+    });
+
+    if (image.length === 0) {
+        throw new Error(`No image found for ${generatorName}`);
+    } else if (image.length > 1) {
+        throw new Error(`Multiple images found at latest tag for ${generatorName}`);
+    }
+
+    // This assumes we have a label of the form version=x.y.z
+    // specifically adding a label to do this to be able to more easily get the version without regex
+    const generatorVersion = image[0]?.Labels["version"];
```

Comment:
assuming this means folks will get upgraded to `rc` versions?

Reply:
chatted live, the CI file makes it so we only tag "latest" if it's not an RC
then this file pulls "latest", grabs the version off of it, and uses that to upgrade the generator